corrai/tsgenerator.py

```diff
@@ -1000,3 +1003,61 @@ def get_dataframe(self, freq="T"):
         df = df.shift(-1)
         df = df.ffill()
         return df.resample(freq).mean()
+
+
+def calculate_power(df, deltaT, Cp):
+    """
+    Calculate power consumption based on given DataFrame.
+
+    Parameters:
+        df (DataFrame): DataFrame containing the data for power calculation.
+        The flow rates of water should be in either L/min or L/h.
+        deltaT (float): Temperature difference (in degrees Celsius).
+        Cp (float): Specific heat capacity of the fluid (in J/(kg·°C)).
+
+    Returns:
+        DataFrame: DataFrame containing the calculated power consumption
+        for each column in the input DataFrame, expressed in kW.
+
+    Raises:
+        ValueError: If the sampling frequency of the DataFrame
+        is not 'H' (hour) or 'T' (minute).
+    """
+    df_powers = df * Cp * deltaT / 3.6e6
```

Comment:
Il y a pas un problème d'unité ? L/min ou L/h * J/kg.K

Reply:
fixed it ! merci